pilot/pkg/model/push_context.go

```diff
@@ -1953,11 +1966,16 @@ func (ps *PushContext) initEnvoyFilters(env *Environment) {
 		return in < jn
 	})
 
-	ps.envoyFiltersByNamespace = make(map[string][]*EnvoyFilterWrapper)
 	for _, envoyFilterConfig := range envoyFilterConfigs {
-		efw := convertToEnvoyFilterWrapper(&envoyFilterConfig)
-		if _, exists := ps.envoyFiltersByNamespace[envoyFilterConfig.Namespace]; !exists {
-			ps.envoyFiltersByNamespace[envoyFilterConfig.Namespace] = make([]*EnvoyFilterWrapper, 0)
+		key := ConfigKey{Kind: kind.EnvoyFilter, Namespace: envoyFilterConfig.Namespace, Name: envoyFilterConfig.Name}
+		var efw *EnvoyFilterWrapper
+		if changed.Contains(key) {
+			efw = convertToEnvoyFilterWrapper(&envoyFilterConfig)
+		} else if prev, ok := previous[key]; ok && features.OptimizedConfigRebuild {
+			// reuse the previous EnvoyFilterWrapper if it exists and optimized config rebuild is enabled
```

Comment:
comment this envoyfilter  is  already exist and not changed

Reply:
done